src/rpg_database/character_sheet/Race.java

```diff
@@ -2,18 +2,56 @@
 
 import static rpg_database.character_sheet.common.CharacterSheetCommon.generateEnumText;
 
-public enum Race {
+import java.util.HashSet;
 
-	HUMAN, ELF, DWARF, QUNARI;
+import rpg_database.character_sheet.exceptions.CharacterSheetException;
+import rpg_database.character_sheet.interfaces.CustomSetter;
+
+public enum Race implements CustomSetter<Race> {
+
+	HUMAN(10), ELF(12), DWARF(8), QUNARI(10);
 
 	private final String text;
+	private final int speed;
 
-	private Race() {
+	private Race(int speed) {
 		this.text = generateEnumText(this.name());
+		this.speed = speed;
+	}
+
+	public int getBaseSpeed() {
+		return this.speed;
 	}
 
 	@Override
 	public String toString() {
 		return text;
 	}
+
+	@Override
+	public Class<Race> getImplementingClass() {
+		return Race.class;
+	}
+
+	@Override
+	public void setSelfInSheet(CharacterSheet characterSheet) {
+		HashSet<Race> allowedRaces = getAllowedRacesForBackground(getActaulBackground(characterSheet));
+		if (allowedRaces.size() > 1)
+			if (allowedRaces.contains(this))
+				characterSheet.characterData.put(Fields.RACE, this);
+			else
+				throw new CharacterSheetException(String.format("%s does not offer %s race for choicing!", getActaulBackground(characterSheet)
+						.toString(), this.text));
+		else
+			throw new CharacterSheetException("The race can not be set manually!");
+	}
+
+	private HashSet<Race> getAllowedRacesForBackground(Background background) {

```

Comment:
Ez is kissé overkill. Egy X.Y().Z() esetén ez jogos, de itt írsz egy metódust ahelyett hogy inline meg kelljen hívnod azt a metódust ami ugyanezt tenné. Valszeg azért csináltad így, mert a lejjebbiből metódust csináltál és így akartad összetömöríteni a kódot, de ott rosszul indultál ugye ki

Reply:
ok